From b8c4d2106f655d7d05bf750275b4bc7d2bed2d3d Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:45 -0400
Subject: [fs] nfsd: remove nfs4_lock_state: nfs4_laundromat

Message-id: <1410898849-22126-162-git-send-email-bfields@redhat.com>
Patchwork-id: 95766
O-Subject: [Patch RHEL7 161/165] nfsd: remove nfs4_lock_state: nfs4_laundromat
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: dab6ef2415a45b644ae6015ea9687803a1e967fa
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 3b9a1a6..f5d434c 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -4152,8 +4152,6 @@ nfs4_laundromat(struct nfsd_net *nn)
  time_t cutoff = get_seconds() - nn->nfsd4_lease;
  time_t t, new_timeo = nn->nfsd4_lease;
 
- nfs4_lock_state();
-
  dprintk("NFSD: laundromat service - starting\n");
  nfsd4_end_grace(nn);
  INIT_LIST_HEAD(&reaplist);
@@ -4221,7 +4219,6 @@ nfs4_laundromat(struct nfsd_net *nn)
  spin_unlock(&nn->client_lock);
 
  new_timeo = max_t(time_t, new_timeo, NFSD_LAUNDROMAT_MINTIMEOUT);
- nfs4_unlock_state();
  return new_timeo;
 }
 
-- 
1.7.1